WikiWatershed/model-my-watershed
# MapShed BMPs: Reclassify Developed Land Categories, and other adjustments

## Overview

This reassigns NLCD 21 Open Space to go to Low Density Residential instead of Low Density Mixed, and updates the impervious percentage values for all urban land use types. Also upgrades GWLF-E, which in https://github.com/WikiWatershed/gwlf-e/pull/84 renames Low Density Residential to Low Density Open Space and adds it to the output. It also removes Other Upland Areas from the output.

Connects #2937 

### Demo

![image](https://user-images.githubusercontent.com/1430060/44693017-9c788c00-aa33-11e8-8d5a-0df13fa84d5c.png)

### Notes

Once this is approved, I will merge the GWLF-E PR and make a new release, and update the final commit here to point to that new release instead of the PR branch.

It was easier to remove "Other Upland Areas" from gwlf-e than to remove it by hand in all the various places in MMW (model results, compare view, CSV exports). This does change the output of gwlf-e, and we'll have to bump up the version number accordingly.

This new value won't show up in Subbasin because SRAT doesn't have a key for this. If they ever add support, we'll have to add that mapping here:

https://github.com/WikiWatershed/model-my-watershed/blob/5260b2e1fd7ce6dbc77e10dc4834b246e03513ff/src/mmw/mmw/settings/gwlfe_settings.py#L623-L638

https://github.com/WikiWatershed/model-my-watershed/blob/5260b2e1fd7ce6dbc77e10dc4834b246e03513ff/src/mmw/mmw/settings/gwlfe_settings.py#L640-L656

## Testing Instructions

* Check out this branch
* Install the latest version of `gwlf-e` in both `app` and `worker` by doing the following:

      $ vagrant ssh app -c 'sudo -H pip uninstall gwlf-e && cd /opt/app && sudo -H pip install -r requirements/development.txt && sudo restart mmw-app'
      $ vagrant ssh worker -c 'sudo -H pip uninstall gwlf-e && cd /opt/app && sudo -H pip install -r requirements/development.txt && sudo service celeryd restart'

* Go to [:8000](http://localhost:8000) and make a new MapShed project.
* Go to Water Quality results. Ensure you see Low-Density Open Space results. Ensure you don't see Other Upland Area results.
* Generate a GMS file. Ensure you see Ld_Open_Space in the file and not Ld_Residential.
* Open Compare View. Ensure the land use categories there match what you saw in model results.
